src/rest.cpp

```diff
@@ -300,7 +300,7 @@ static bool rest_mempool_info(HTTPRequest* req, const std::string& strURIPart)
 
     switch (rf) {
     case RetFormat::JSON: {
-        UniValue mempoolInfoObject = MempoolInfoToJSON(::mempool);
+        UniValue mempoolInfoObject = MempoolInfoToJSON(EnsureMemPool());
```

Comment:
Are we really allowed to throw `JSONRPCError`s in the rest code? I don't see it happening anywhere else. Where does it get caught?

Reply:
good point. Changed it to return `404` when the mempool is not found.